### PR TITLE
Strings docs

### PR DESCRIPTION
**Pull request by Hugheth**

### CODE REVIEW

File: src/Strings.lua [hunk 1]
```diff
@@ -1,48 +1,346 @@
...
-function Strings.leftPad(string, length, character) --: (string, number, string) => string
-	return string.rep(character or " ", length - #string) .. string
+--[[
+	Encodes _str_ for use in a url, for example as a query parameter of a url.
+	@trait Chainable
+	@example
+		_.encodeUrlComponent("https://Egg+Fried Rice!?")
+		--> "https%3A%2F%2FEgg%2BFried%20Rice!%3F"
+]]
+function Strings.encodeUrlComponent(str)
```
Comment (ckdavid) [inline src/Strings.lua]: How does this differ from https://developer.roblox.com/en-us/api-reference/function/HttpService/UrlEncode?

File: src/Strings.lua [hunk 2]
```diff
@@ -1,48 +1,346 @@
...
+			chocolatey = true
+		})) --> "?biscuits=hobnobs&time=11&chocolatey=true"
+]]
+function Strings.encodeQueryString(query)
+	assert(t.table(query))
+	local fields =
+		Tables.mapValues(
+		query,
+		function(value, key)
+			return key .. "=" .. Strings.encodeUrlComponent(tostring(value))
```
Comment (ckdavid) [inline src/Strings.lua]: Shouldn't this also be URL encoding the keys?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Setup
+
+Clone the repo locally with `git clone git@github.com:CodeKingdomsTeam/rodash.git`.
+
+| Name                 | Notes                                                          |
+| -------------------- | -------------------------------------------------------------- |
+| `setup.sh`           | Installs Lua 5.1 and dependencies locally.                     |
```
[the remark this answers is not in the document]
Comment (Hugheth) [reply CONTRIBUTING.md]: I've added step-by-step instructions for Windows. Git for Windows installs Git BASH, so I encourage them to use that for sh rather than maintaining separate DOS style bat files

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+function onlyLocalPlayerHasSpoon()
+
+	local playersWithSpoon = _.filter(game.Players, function( player )
+		return _.some(player.Backpack, function( tool )
+			return _.endsWith(tool.Name, "Spoon")
+		end)
+	end)
+
+	return _.matches(playersWithSpoon, {game.Players.LocalPlayer})
```
[the remark this answers is not in the document]
Comment (Hugheth) [reply docs/index.md]: this is being added to the Tables PR at a later stage

File: spec/Strings.spec.lua [hunk 5]
```diff
@@ -4,43 +4,147 @@ describe(
...
+			function()
+				it(
+					"from snake-case",
+					function()
+						assert.are.same("strong-stilton", Strings.kebabCase("__STRONG_STILTON__"))
 					end
 				)
 				it(
-					"splits with empty delimiter",
+					"from kebab-case",
```
Comment (ckdavid) [inline spec/Strings.spec.lua]: `camelCase`

File: spec/Strings.spec.lua [hunk 6]
```diff
@@ -4,43 +4,147 @@ describe(
...
 				)
 			end
 		)
 
 		describe(
-			"Trim",
+			"titleCase",
+			function()
+				it(
+					"words",
```
Comment (ckdavid) [inline spec/Strings.spec.lua]: What's the reason for deviating from the format of descriptions above (i.e. "from X")?

File: src/Strings.lua [hunk 7]
```diff
@@ -1,48 +1,396 @@
...
+	local remainder = padLength % #prefix
+	local repetitions = (padLength - remainder) / #prefix
+	return string.rep(prefix or " ", repetitions) .. prefix:sub(1, remainder) .. str
 end
 
-function Strings.rightPad(string, length, character) --: (string, number, string) => string
-	return string .. string.rep(character or " ", length - #string)
+--[[
+	Makes a string of `length` from `str` by repeating characters from `suffix` at the end of the string.
+	@example _.leftPad("toast", 6) --> "toast "
```
Comment (ckdavid) [inline src/Strings.lua]: 🚨 copy & paste alert 🚨

File: src/Tables.lua [hunk 8]
```diff
@@ -192,7 +242,17 @@ function Tables.keyBy(source, handler) --: <T, K extends Key>(T[], (T => K) => I
 	return result
 end
 
-function Tables.groupBy(source, handler) --: <T extends Iterable<K,V>, I extends Key>((value: T, key: K) => I) => Iterable<I, Iterable<K,V>>)
+--[[
+	Summary ends with a period.
```
Comment (ckdavid) [inline src/Tables.lua]: I assume this is all going to be fixed in a different PR.

File: spec/Strings.spec.lua (hunk 5 above)
Comment (Hugheth) [reply spec/Strings.spec.lua]: Not sure what you mean by this? I'm using `kebab-case` as that's the natural english way of writing it, and `camelCase` above as that's the method name

Comment (ckdavid) [reply spec/Strings.spec.lua]: It should be `from camel-case` (`strongStilton`).
